### PR TITLE
Create all resources necessary for the agent to run on Vintage MCs

### PR DESCRIPTION
**Pull request by QuentinBisson**
## Checklist

Trying to enable the agent on Vintage MCs so we can have the same behavior everywhere

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

**Comment by QuentinBisson** [comment]
Testing with https://github.com/giantswarm/aws-app-collection/pull/195/files

**Comment by QuentinBisson** [comment]
image

**Comment by QuentinBisson** [comment]
kube-controller-manager is missing the component label:

  selector:
    component: kube-controller-manager


```sh
k get pods -n kube-system k8s-controller-manager-ip-10-0-5-145.eu-west-1.compute.internal -oyaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    kubernetes.io/config.hash: ca8acf5c1621bc4adfd3846627a151d0
    kubernetes.io/config.mirror: ca8acf5c1621bc4adfd3846627a151d0
    kubernetes.io/config.seen: "2022-11-15T16:28:03.066675832Z"
    kubernetes.io/config.source: file
    kubernetes.io/psp: calico-kube-controllers-psp
    scheduler.alpha.kubernetes.io/critical-pod: ""
  creationTimestamp: "2022-11-15T16:28:04Z"
  name: k8s-controller-manager-ip-10-0-5-145.eu-west-1.compute.internal
  namespace: kube-system
  ownerReferences:
  - apiVersion: v1
    controller: true
    kind: Node
    name: ip-10-0-5-145.eu-west-1.compute.internal
    uid: 7623e5de-485d-40b5-b7cf-4a225efe214e
  resourceVersion: "928491767"
  uid: 1427acc3-a1f3-484d-aa28-473294466cd4

```

**Comment by QuentinBisson** [comment]
Same for the scheduler

**Comment by QuentinBisson** [comment]
Vintage k8s scheduler is using the wrong port 10257 when it should use 10259 as is the case in CAPI:
      httpGet:
        path: /healthz
        port: 10259
        scheme: HTTPS


This needs to be changed

**Comment by QuentinBisson** [comment]
Tested with https://github.com/giantswarm/aws-app-collection/pull/195 https://github.com/giantswarm/observability-bundle/pull/21 successfully

**Comment by hervenicol** [comment]
I'm disturbed reading the code: it seems that it does much more than remote writes.

I feel like it does now rely on the prometheus-agent being deployed.
Which includes stuff like stop prometheus from scraping a few targets (apiserver, controllermanager, scheduler) on vintage MCs.

I can't accept nor deny this PR, as I can barely understand it and won't spot errors :grimacing:

File: service/key/key.go [hunk 1]
```diff
@@ -152,13 +152,15 @@ func RemoteWriteAuthenticationAnnotations() map[string]string {
...
-func RemoteWriteAPIEndpointConfigSecretNameAndNamespace(cluster metav1.Object, provider string) (string, string) {
+func RemoteWriteAPIEndpointConfigSecretNameAndNamespace(cluster metav1.Object, installation string, provider string) (string, string) {
 	name := RemoteWriteAPIEndpointConfigSecretName
 	namespace := ClusterID(cluster)
 
 	if IsCAPIManagementCluster(provider) {
 		name = ClusterID(cluster) + "-" + name
 		namespace = cluster.GetNamespace()
+	} else if IsManagementCluster(installation, cluster) {
+		namespace = "giantswarm"
```
Comment (marieroque) [inline service/key/key.go:163]: Maybe a const for this kind of value

File: helm/prometheus-meta-operator/Chart.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-apiVersion: v1
+apiVersion: v2
```
Comment (marieroque) [inline helm/prometheus-meta-operator/Chart.yaml:1]: API has breaking changes ?

Comment (QuentinBisson) [reply helm/prometheus-meta-operator/Chart.yaml:1]: That's helm but I can remove that tbh

Comment (QuentinBisson) [reply helm/prometheus-meta-operator/Chart.yaml:1]: There is no breaking change

File: service/controller/managementcluster/resource.go [hunk 3]
```diff
@@ -349,6 +350,9 @@ func newResources(config resourcesConfig) ([]resource.Interface, error) {
 		alertmanagerConfigResource,
 		heartbeatWebhookConfigResource,
 		alertmanagerWiringResource,
+		remoteWriteAPIEndpointConfigSecretResource,
```
Comment (QuentinBisson) [inline service/controller/managementcluster/resource.go:353]: Adding the resources to vintage MCs also

File: service/controller/resource/monitoring/prometheus/resource.go [hunk 4]
```diff
@@ -340,13 +340,12 @@ func toPrometheus(ctx context.Context, v interface{}, config Config) (metav1.Obj
 			},
 		}
 
-		allMonitorSelector := []metav1.LabelSelectorRequirement{}
-		if key.IsCAPIManagementCluster(config.Provider) {
-			// We do not discover the service monitors discovered by the agent running on the management cluster
-			allMonitorSelector = append(allMonitorSelector, metav1.LabelSelectorRequirement{
+		// We do not discover the service monitors discovered by the agent running on the management cluster
```
Comment (QuentinBisson) [inline service/controller/resource/monitoring/prometheus/resource.go:343]: Support both MC types

**Comment by QuentinBisson** [comment]
I also refactored isInCluster to isManagementCluster because it made no sense

**Pull request merged**